### PR TITLE
ServiceUnavailableExceptionMessageImprovement

### PR DESCRIPTION
**Pull request by xinlian12**
Addresses: https://github.com/Azure/azure-sdk-for-java/issues/11674

1. Include real exception message when create ServiceUnavailableException
2. Renaming parameter name to be *Properties for *Properties class type. For example: 
triggerSettings -> triggerProperties.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/BridgeInternal.java [hunk 1]
```diff
@@ -485,4 +489,9 @@ public static CosmosUser createCosmosUser(CosmosAsyncUser asyncUser, CosmosDatab
     public static ConsistencyLevel fromServiceSerializedFormat(String consistencyLevel) {
         return ConsistencyLevel.fromServiceSerializedFormat(consistencyLevel);
     }
+
+    @Warning(value = INTERNAL_USE_ONLY_WARNING)
+    public static CosmosException createServiceUnavailableException(Exception innerException) {
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/BridgeInternal.java:494]: I think generally, it is better to create more specific exception compared to CosmosExceptions. 
(Another example is when create BadRequestExceptions)

Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/BridgeInternal.java:494]: Yes, it is. We can incorporate those changes after GA.

**Comment by xinlian12** [comment]
Run the following local tests:
GoneAndRetryWithRetryPolicyTest
CosmosExceptionTest

**Pull request merged**